backend/src/main/java/com/team4/backend/service/MonitorService.java

```diff
@@ -0,0 +1,24 @@
+package com.team4.backend.service;
+
+import com.team4.backend.model.Monitor;
+import com.team4.backend.repository.MonitorRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+@Service
+public class MonitorService {
+
+    @Autowired
+    private MonitorRepository monitorRepository;
+
+    public Mono<Monitor> findMonitorByEmail(String email) {
+        return monitorRepository.findByEmail(email)
+                .filter(Objects::nonNull)
+                .switchIfEmpty(Mono.error(new ResponseStatusException(HttpStatus.BAD_REQUEST, "Can't find monitor with this email!")));
+    }
```

Comment:
Do not handle Http related objects in service, create new custom exception and use with Mono.error() (See Student controller && service for examples)